src/main/scala/language.scala

```diff
@@ -0,0 +1,51 @@
+package piconot
+import scalafx.application.JFXApp
+import picolib.semantics._
+/**
+ * @author justisallen
+ */
+trait language extends JFXApp{
+ 
+  val blank = Surroundings(Anything,Anything,Anything,Anything)
+  val No = Surroundings(Open,Anything,Anything,Anything)
+  val Nx = Surroundings(Blocked,Anything,Anything,Anything)
+  val So = Surroundings(Anything,Anything,Anything,Open)
+  val Sx = Surroundings(Anything,Anything,Anything,Blocked)
+  val Eo = Surroundings(Anything,Open,Anything,Anything)
+  val Ex = Surroundings(Anything,Blocked,Anything,Anything)
+  val Wo = Surroundings(Anything,Anything,Open,Anything)
+  val Wx = Surroundings(Anything,Anything,Blocked,Anything)
+  val X = StayHere
+  val N = North
+  val S = South
+  val E = East
+  val W = West
+
+  
+  implicit class rule(val state: Int)(val posStatus: Surroundings*) {
```

Comment:
If it was an attempt to get rid of the `rule` keyword, then I believe you could have done this by making an `apply` method for the surroundings instead of making a curried constructor. Then you could have said `0 (Wo) -> (W)` for the first rule of `Empty`, and similar for the others.